packages/x-components/src/x-modules/facets-next/components/lists/filters-injection.mixin.ts

```diff
@@ -0,0 +1,72 @@
+import { Filter } from '@empathyco/x-types-next';
+import Vue from 'vue';
+import { Prop } from 'vue-property-decorator';
+import Component from 'vue-class-component';
+import { HierarchicalFilter, isHierarchicalFilter } from '@empathyco/x-types-next';
+import { XInject } from '../../../../components/decorators/injection.decorators';
+import { isArrayEmpty } from '../../../../utils/array';
+
+/**
+ * Mixin to share filters injection logic.
+ *
+ * @public
+ */
+@Component
+export default class FiltersInjectionMixin extends Vue {
+  /**
+   * The list of filters to be rendered as slots.
+   *
+   * @public
+   */
+  @Prop()
+  protected filters!: Filter[];
+
+  @Prop({ required: false })
+  protected parentId!: Filter['id'];
+
+  /** The injected filters array.
+   *
+   * @public
+   */
+  @XInject('filters')
+  public injectedFilters!: Filter[];
+
+  /**
+   * The prop or injected filters array, filtered by parentId if they are
+   * {@link HierarchicalFilter}.
+   *
+   * @returns An array of filters.
+   *
+   * @internal
+   */
+  protected get renderedFilters(): Filter[] {
+    return this.filterByParentId(this.propOrInjectedFilters);
+  }
+
+  /**
+   * An array of filters formed by those that are passed through prop or injected.
+   *
+   * @returns An array of filters.
+   *
+   * @internal
+   */
+  protected get propOrInjectedFilters(): Filter[] {
+    return (
+      this.filters ??
+      this.injectedFilters ??
+      //TODO: add here logger
```

Comment:
Why are we not using the logger here (just curiosity)?

Reply:
Because we didn't start using the logger in XComponents. All these TODOs are for that moment we will include the logger.